packages/arb-token-bridge-ui/src/components/common/AddCustomChain.tsx

```diff
@@ -164,10 +166,18 @@ function saveOrbitConfigToLocalStorage(data: OrbitConfig) {
   )
 }
 
-function mapOrbitConfigToOrbitChain(data: OrbitConfig): ChainWithRpcUrl {
+async function mapOrbitConfigToOrbitChain(
+  data: OrbitConfig
+): Promise<ChainWithRpcUrl> {
+  const rollup = RollupAdminLogic__factory.connect(
+    data.coreContracts.rollup,
+    getProviderForChainId(data.chainInfo.parentChainId)
+  )
+  const confirmPeriodBlocks =
+    (await rollup.confirmPeriodBlocks()).toNumber() ?? 150
   return {
     chainID: data.chainInfo.chainId,
-    confirmPeriodBlocks: 150,
+    confirmPeriodBlocks,
```

Comment:
only get the accurate confirm period blocks for newly added chains

old custom chains stick to 150